packages/opentelemetry-core/test/context/B3Propagator.test.ts

```diff
@@ -98,127 +158,350 @@ describe('B3Propagator', () => {
+          defaultGetter
+        );
+        const extractedSpanContext = getExtractedSpanContext(context);
+
+        assert.deepStrictEqual(extractedSpanContext, {
+          spanId: 'b7ad6b7169203331',
+          traceId: '0af7651916cd43dd8448eb211c80319c',
+          isRemote: true,
+          traceFlags: TraceFlags.SAMPLED,
+        });
+        assert.equal(context.getValue(DEBUG_FLAG_KEY), undefined);
+        assert.equal(context.getValue(PARENT_SPAN_ID_KEY), undefined);
+      });
+
+      it('should extract context of a sampled span from carrier when sampled is mentioned as boolean false flag', () => {
+        carrier[X_B3_TRACE_ID] = '0af7651916cd43dd8448eb211c80319c';
+        carrier[X_B3_SPAN_ID] = 'b7ad6b7169203331';
+        carrier[X_B3_SAMPLED] = false;
+        const context = b3Propagator.extract(
+          Context.ROOT_CONTEXT,
+          carrier,
+          defaultGetter
+        );
+        const extractedSpanContext = getExtractedSpanContext(context);
+
+        assert.deepStrictEqual(extractedSpanContext, {
+          spanId: 'b7ad6b7169203331',
+          traceId: '0af7651916cd43dd8448eb211c80319c',
+          isRemote: true,
+          traceFlags: TraceFlags.NONE,
+        });
+        assert.equal(context.getValue(DEBUG_FLAG_KEY), undefined);
+        assert.equal(context.getValue(PARENT_SPAN_ID_KEY), undefined);
       });
     });
 
-    it('should extract context of a sampled span from carrier when sampled is mentioned as boolean true flag', () => {
-      carrier[X_B3_TRACE_ID] = '0af7651916cd43dd8448eb211c80319c';
-      carrier[X_B3_SPAN_ID] = 'b7ad6b7169203331';
-      carrier[X_B3_SAMPLED] = true;
-      const extractedSpanContext = getExtractedSpanContext(
-        b3Propagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
-      );
+    describe('when debug flag is valid', () => {
+      it('should extract context of a debug span from carrier when debug flag is valid', () => {
+        carrier[X_B3_TRACE_ID] = '0af7651916cd43dd8448eb211c80319c';
+        carrier[X_B3_SPAN_ID] = 'b7ad6b7169203331';
+        carrier[X_B3_FLAGS] = '1';
+        const context = b3Propagator.extract(
+          Context.ROOT_CONTEXT,
+          carrier,
+          defaultGetter
+        );
+        const extractedSpanContext = getExtractedSpanContext(context);
+
+        assert.deepStrictEqual(extractedSpanContext, {
+          spanId: 'b7ad6b7169203331',
+          traceId: '0af7651916cd43dd8448eb211c80319c',
+          isRemote: true,
+          traceFlags: TraceFlags.SAMPLED,
+        });
+        assert.strictEqual(context.getValue(DEBUG_FLAG_KEY), '1');
+        assert.equal(context.getValue(PARENT_SPAN_ID_KEY), undefined);
+      });
+    });
 
-      assert.deepStrictEqual(extractedSpanContext, {
-        spanId: 'b7ad6b7169203331',
-        traceId: '0af7651916cd43dd8448eb211c80319c',
-        isRemote: true,
-        traceFlags: TraceFlags.SAMPLED,
+    describe('when debug flag is invalid', () => {
```

Comment:
Yeah, I could do that if that's preferable. I personally think that the extra describe is kind of superfluous when there's only one test case in it, which is what most of these tests will end up having. It seems like a lot of boilerplate for little benefit. Plus, it also seems weird that none of the already existing tests won't have the structure (unless you think  Ishould do those too).

Reply:
I also don't see this pattern in the other propagation tests, which adds to the hesitation, but I see what you're saying. Some of my its are a bit verbose.

Reply:
I think maybe I can just reanme something like ```it(should extract context of a sampled span from carrier when sampled is mentioned as boolean true flag)````
```it(should extract when sampled is true)```

Reply:
Our unit tests are not perfect for sure, but with a small effort, whenever we do something in certain part of code, there is always some place to make some small improvements. The unit test by definition should test one single thing, so whenever something fails just by looking at description you should be sure what has just happened and it is also a nice way of documenting the causes and results to understand the whole flow better. This is particularly interesting when when looking later on tree structure of unit tests. I'm not forcing you to do this, I just wanted to encourage for a refactoring some description so you can group some results / cases better / to be more readable (imho) and see what different edge cases you have for certain "if/when" condition. We already have different way of "styling" unit tests depends on personal preference from different people. I personally prefer more "describe" instead of having unit test with long description when "if/when is quite often at the end of sentence describing the case so you have to read to the end of the description to be able to find what you are looking for.

Reply:
fasho, sounds good to me. I'll do it. It does read a little nicer, just wasn't sure about the reasoning. Thanks for the explanation.